Lib/typing.py

```diff
@@ -183,14 +187,14 @@ def _type_repr(obj):
 
 
 def _collect_type_vars(types):
-    """Collect all type variable contained in types in order of
-    first appearance (lexicographic order). For example::
+    """Collect all type variable and parameter specification variables contained
```

Comment:
Thoughts about this?

Reply:
I changed all the methods that check for `(TypeVar, ParamSpec)` to just check for `_TypeVarLike` instead. Which means I can probably get rid of some tests for `ParamSpec` since it uses the exact same `__reduce__`, `__repr__`, `__ror__` etc. methods as `TypeVar`, and `TypeVar` already has tests for them.